yt_dlp/extractor/arte.py

```diff
@@ -46,154 +49,150 @@ class ArteTVIE(ArteTVBaseIE):
-                r'V{0}-STM{0}$'.format(l),
-                # original version in requested language, with partial subtitles in different language
-                r'VO{0}-ST(?!{0}).+?$'.format(l),
-                # original version in requested language, with subtitles for the deaf and hard-of-hearing in different language
-                r'VO{0}-STM(?!{0}).+?$'.format(l),
-                # original version in different language, with partial subtitles in requested language
-                r'VO(?:(?!{0}).+?)?-ST{0}$'.format(l),
-                # original version in different language, with subtitles for the deaf and hard-of-hearing in requested language
-                r'VO(?:(?!{0}).+?)?-STM{0}$'.format(l),
-                # original version in different language, without subtitles
-                r'VO(?:(?!{0}))?$'.format(l),
-                # original version in different language, with partial subtitles in different language
-                r'VO(?:(?!{0}).+?)?-ST(?!{0}).+?$'.format(l),
-                # original version in different language, with subtitles for the deaf and hard-of-hearing in different language
-                r'VO(?:(?!{0}).+?)?-STM(?!{0}).+?$'.format(l),
-            )
-
-            for pref, p in enumerate(PREFERENCES):
-                if re.match(p, versionCode):
-                    lang_pref = len(PREFERENCES) - pref
-                    break
-            else:
-                lang_pref = -1
-            format_note = '%s, %s' % (f.get('versionCode'), f.get('versionLibelle'))
-
-            media_type = f.get('mediaType')
-            if media_type == 'hls':
-                m3u8_formats = self._extract_m3u8_formats(
-                    format_url, video_id, 'mp4', entry_protocol='m3u8_native',
-                    m3u8_id=format_id, fatal=False)
-                for m3u8_format in m3u8_formats:
-                    m3u8_format.update({
+        subtitles = {}
+
+        for stream in config['data']['attributes']['streams']:
+            # XXX: the fallback when there are none looks reasonable, but
+            # what if there is more than one version? the host’s JS
+            # does not handle it well either
+
+            streamVersion = try_get(stream, lambda s: s['versions'][0]) or {}
+            streamVersionCode = streamVersion.get('eStat', {}).get('ml5')
+
+            lang_pref = -1
+            m = self.__VERSION_CODE_RE.match(streamVersionCode)
+            if m:
+                lc = self.__LANG_MAP.get(lang)
+                lang_pref = sum(
+                    pref << (5 - i)
+                    for i, pref in enumerate((
+                        m.group('vlang') == lc,
+                        not not m.group('vo'),
+                        m.group('stlang') == lc,
+                        not m.group('st'),
+                        not m.group('stm'),
+                    ))
+                )
```

Comment:
The `5-i` is asking for mistakes if the list expands
```suggestion
                lang_pref = int(''.join('01'[x] for x in (
                    m.group('vlang') == lc,
                    bool(m.group('vo')),
                    m.group('stlang') == lc,
                    not m.group('st'),
                    not m.group('stm'),
                )), 2)
```

Reply:
I still like mine better, but that's acceptable